tests/test_register.py

```diff
@@ -0,0 +1,162 @@
+"""Testing all things registration through the API."""
+import json
+import unittest
+from app import create_app, db
+from app.models import User
+from flask import url_for
+
+
+class ApiRegistration(unittest.TestCase):
+    """Test user registration through the API."""
+
+    def setUp(self):
+        """Setup app for testing before each test case."""
+        self.app = create_app("testing")
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+        db.create_all()
+        self.client = self.app.test_client(use_cookies=True)
+
+    def tearDown(self):
+        """Delete app and db instances after each test case."""
+        db.session.remove()
+        db.drop_all()
+        self.app_context.pop()
```

Comment:
Since you perform this app creation in (I think) all your tests. You might consider having a Base class that gets this setup for other tests to inherit and consume.